### PR TITLE
Maintenance: Refactor Popover/Tooltip components

### LINKED ISSUE #1812: Refactor Popover/Tooltip components
Originally posted by @chmanie in #1816:

We have a quite involved [Popover component](https://github.com/JoinColony/colonyCDapp/blob/master/src/components/shared/Popover/Popover.tsx) that is only used once. Maybe we can investigate whether it is still necessary to have this (also to keep things consistent). Maybe also investigate the Tooltip component (which is used twice) in the same directory.

### PR DESCRIPTION
**Pull request by iamsamgibbs**
## Description

This PR removes a [Popover component](https://github.com/JoinColony/colonyCDapp/blob/master/src/components/shared/Popover/Popover.tsx) that was only used once.

It also removes the tooltip component from the same directory.

As part of this I tidied up the NativeTokenPill component where the popover component was in use to add hover and active states, and made the token popup function as a modal on mobile to be consistent with the over token popups across the app.

## Testing

Check the NativeTokenPill on the dashboard and colony admin page to ensure they have not regressed.

Screenshot 2024-03-26 at 14 28 48

Screenshot 2024-03-26 at 14 28 59

I'm not sure how which transactions get the `MessageCard` component to show - for the sake of fixing this I adjusted the `TransactionList` component to manually get it to show.

Screenshot 2024-03-26 at 15 36 20

## Diffs

**Changes** 🏗

* NativeTokenPill no longer uses popover component
* MessageCardStatus no longer uses old tooltip component

**Deletions** ⚰️

* Popover component
* Tooltip component

Resolves #1812

### CODE REVIEW

File: src/components/v5/common/NativeTokenPill/NativeTokenPill.tsx [hunk 1]
```diff
@@ -20,29 +26,81 @@ const NativeTokenPill = ({
...
+        >
+          {token.symbol}
+        </span>
+        {isLocked && (
+          <Tooltip
+            tooltipContent={
+              <span>{formatText({ id: 'tooltip.lockedToken' })}</span>
+            }
+          >
+            <LockKey size={11} className="ml-0.5" />
```
Comment (rdig) [inline src/components/v5/common/NativeTokenPill/NativeTokenPill.tsx:74]: As I commented on @bassgeta 's #2066, I really wish we'd had `enums` for sizes

**Review by rdig — APPROVED** [review]
All good from my perspective, nice work! I've picked up a couple of minor things, which you should look into, but overall this is good to go.

The token popovers didn't seem to have regressed

Screenshot from 2024-03-26 21-05-04
Screenshot from 2024-03-26 21-05-28

However, the tooltips on the the motion pills are being cut out to the right of the view port _(although not sure if it's related to this or not)_

Screenshot from 2024-03-26 21-07-15

Regarding your comment

> I'm not sure how which transactions get the MessageCard component to show - for the sake of fixing this I adjusted the TransactionList component to manually get it to show.

Anything that does message signing: voting, revealing, metatxs _(although I actually think we've removed it from showing up at all)_

**Comment by melyndav** [comment]
@rdig @iamsamgibbs just an FYI. I opened an issue for the desktop tooltips on motion pills [here.](https://github.com/JoinColony/colonyCDapp/issues/2062).

**Review by melyndav — APPROVED** [review]
Heya @iamsamgibbs nice work on this one!

Token popovers look like they should. We discussed the tooltip scope, so I'm happy to see this one go live! Thanks again. :)

**Pull request merged**